lib/include/tree_sitter/api.h

```diff
@@ -551,6 +551,13 @@ TSStateId ts_node_next_parse_state(TSNode self);
  */
 TSNode ts_node_parent(TSNode self);
 
+/**
+ * Get the node's immediate child that contains a specified subnode.
+ * Prefer this function over ts_node_parent()
+ * when traversing the parent nodes of the subnode.
+ */
+TSNode ts_node_child_with_subnode(TSNode self, TSNode subnode);
```

Comment:
This also feels like a decent name. You're looking for the "next child on the unique tree _path_ between node A and node B"

```c
TSNode ts_node_child_on_path(TSNode self, TSNode node);
```

Reply:
```c
/**
 * Get the node's immediate parent.
 * Prefer [`ts_node_child_on_path`] for 
 * iterating over the node's parents.
 */
TSNode ts_node_parent(TSNode self);

/**
 * Get the node's child on the path to `descendant`.
 * `descendant` itself is never returned.
 */
TSNode ts_node_child_on_path(TSNode self, TSNode descendant);
```
?

Reply:
How about `ts_node_child_containing_descendant`